includes/class-post-types.php

```diff
@@ -406,6 +406,17 @@ public static function register_post_post_type() {
 				'sanitize_callback' => 'absint',
 			)
 		);
+
+		\register_post_meta(
+			Posts::POST_TYPE,
+			'_activitypub_featured_image_url',
+			array(
+				'type'              => 'string',
+				'single'            => true,
+				'description'       => 'URL of the featured image from ActivityPub attachments.',
+				'sanitize_callback' => 'sanitize_url',
+			)
+		);
```

Comment:
Instead of using custom storage we should add `'thumbnail'` support to the custom post type registration and re-use Core APIs.